main.py

```diff
@@ -0,0 +1,172 @@
+print(f"aUSD/USD had an annualized mean return of {ausd_pair.calculate_mean_return()}")
+# We can disregard the mean return as it is assumed to be 0 for stable coins due to arbitrage
+# We have to take into consideration the standard deviation of the price, which we will
+# add to the standard deviation of BTC
+
+#%%
+# BTC is the debt in the system and if BTC increases in price, the over-collateralization ratio drops
+# To model this, we get the inverse price to have BTC as quote currency
+# If we select the n-th worst trajectories of the inverse price,
+# it's equivalent to the n-th best price appreciation.
+# We choose a 5 year period to get a decent sample period with 3 bull runs & crashes
+quote_currency = Token("usd", "USD")
+base_currency = Token("bitcoin", "BTC")
+
+btc_pair = Token_Pair(base_currency, quote_currency)
+btc_pair.get_prices(start_date="2017-07-01", inverse=True)
+btc_pair.calculate_returns()
+
+print(f"USD/BTC had an annualized std of {btc_pair.returns.std()[0] *365**0.5}")
+print(f"USD/BTC had an annualized mean return of {btc_pair.calculate_mean_return()}")
+
+# %%
+# We adjust the variance by adding the covariance of the aUSD/USD pair.
+corr = ausd_pair.returns.corrwith(btc_pair.returns)
+cov = corr * (btc_pair.returns.std()[0] *365**0.5) * (ausd_pair.returns.std()[0] *365**0.5)
+# daily var scaled to a year, add 2*covariance and take the square root to get the std
+adj_btc_std = (btc_pair.returns.var()[0] *365 + 2 * cov) ** 0.5
+
+# Initialize and run the simulation: Each path represents the price change of the collateral/debt
+# We simulate 10,000 trajectories with a duration of 1 year and 365 trading days
+# and assume a normal distribution (GBM) with the mean and std of bitcoin over the past 5 years
+sim = Simulation(btc_pair, strategy="GBM")
+sim.simulate(steps=365,
+             maturity=1,
+             n_simulations=10_000,
+             initial_value=1,
+             sigma=adj_btc_std[0],
+             mu=btc_pair.calculate_mean_return(type="arithmetic"))
+
+
+# %%
+# Analize the results
+# Initialize the analysis and plot all paths
+simple_analysis = Analysis(sim)
+simple_analysis.plot_returns("Percentage", "Performance", type="line")
+
+# This gives us the secure threshold multiplier, stating that:
+# The collateral/debt ratio will fall below 1 only with a 10% chance in a year or...
+# We should see this happening once every ~10 years.
+# This assumes no premium redeem, additional collateralization or liquidation.
+secure_threshold = simple_analysis.get_threshold_multiplier(alpha=0.90) # <- this can be changed ofc 
+
+print(f"The estimated liquidation threshold is ~{int(secure_threshold * 100)}% of the debt value")
+
+# %%
+# This time we do the same process on a intraday basis
+# We adjust the variance by adding the covariance of the aUSD/USD pair.
+corr = ausd_pair.returns.corrwith(btc_pair.returns)
+cov = corr * btc_pair.returns.std()[0] * ausd_pair.returns.std()[0]
```

Comment:
this should be a utility function that take both returns arrays and returns the correlation and covariance

Reply:
This was a very special use case and I don't know if I would ever need to do this again. If so, I'll make a function as you said.